src/index.js

```diff
@@ -129,15 +133,22 @@ export function controller(RootComponent, controllerGenerators, selectorBundles,
     }
 
     componentWillMount() {
-      if (this.controllerMethods.initialize) { this.controllerMethods.initialize(); }
+      if (this.controllerMethods.initialize) {
+        this.controllerMethods.initialize();
+      }
+    }
+
+    componentWillUnMount() {
+      if (this.controllerMethods.deinitialize) {
+        this.controllerMethods.deinitialize();
+      }
     }
 
     getChildContext() {
       // Rather than injecting all of the RootComponent props into the context,
-      // we only explictly pass selector and controller method props.
-      const selectorProps = R.pick(R.keys(selectorPropTypes), this.props);
-      const childContext = R.merge(selectorProps, this.controllerMethods);
-      return childContext;
```

Comment:
Keeping an explicit return variable makes things easier to debug and also makes the code more readable ("these are the props and methods that will be injected into the child's context")